polyfill/test262-harness.mjs

```diff
@@ -0,0 +1,246 @@
+  // is given in the expected failures file. testDisplayName is a slightly
+  // abbreviated form that we use in logging during the run to make it more
+  // likely to fit on one line. progressDisplayName is what's displayed beside
+  // the progress bar: testDisplayName with the actual test filename cut off,
+  // since the individual tests go by too fast to read anyway.
+  const testRelPath = path.relative('test262/test', testFile);
+  const testDisplayName = testRelPath
+    .replace('built-ins/Temporal/', '')
+    .replace('intl402/Temporal/', '(intl) ')
+    .replace('staging/Temporal/', '(stg) ')
+    .replace('/prototype/', '/p/');
+  const progressDisplayName = path.dirname(testDisplayName);
+  progress.tick(0, { test: progressDisplayName });
+
+  // Time each test individually in order to report if they take longer than
+  // 100 ms
+  const testStart = process.hrtime.bigint();
+
+  // Run the test and log a message above the progress bar if the result is not
+  // what it's supposed to be. This is so that you don't have to wait until the
+  // end to see if your test failed.
+  try {
+    vm.runInContext(testCode, testContext);
+    passed++;
+
+    if (expectedFailures.has(testRelPath)) {
+      unexpectedPasses.push(testRelPath);
+      progress.interrupt(`UNEXPECTED PASS: ${testDisplayName}`);
+    }
+  } catch (e) {
+    if (expectedFailures.has(testRelPath)) {
+      skipped++;
+    } else {
+      failures.push({ file: testRelPath, error: e });
+      progress.interrupt(`FAIL: ${testDisplayName}`);
+    }
+  }
+
+  const testFinish = process.hrtime.bigint();
+  const testTime = testFinish - testStart;
+  if (testTime > 100_000_000n) {
+    longTests.push({ file: testRelPath, ns: testTime });
+  }
+
+  progress.tick(1, { test: progressDisplayName });
+}
+
+// === Print results ===
+
+const finish = process.hrtime.bigint();
+const elapsed = Number(finish - start) / 1_000_000_000;
+const failed = failures.length;
+
+print(color.underline('\nSummary of results:'));
+
+if (failed > 0) {
+  failures.forEach(({ file, error }) => {
+    print(color.yellow(`\n${color.bold('FAIL')}: ${file}`));
+    if (error.constructor.name === 'Test262Error') {
+      print(` \u2022 ${error.message}`);
+    } else {
+      print(util.inspect(error, { colors: color.enabled }));
+    }
+  });
+}
+
+if (unexpectedPasses.length > 0) {
+  print(`\n${color.yellow.bold('WARNING:')} Tests passed unexpectedly; remove them from expected-failures?`);
+  unexpectedPasses.forEach((file) => print(` \u2022  ${file}`));
+}
+
+if (longTests.length > 0) {
+  print('\nThe following tests took a long time:');
+  longTests.forEach(({ file, ns }) => {
+    print(`  ${color.yellow(Math.round(Number(ns) / 1_000_000))} ms: ${file}`);
+  });
+}
+
+print(`\n${total} tests finished in ${color.bold(elapsed.toFixed(1))} s`);
+print(color.green(`  ${passed} passed`));
+print(color.red(`  ${failed} failed`));
+if (skipped > 0) print(color.cyan(`  ${skipped} skipped`));
```

Comment:
I find this variable name and corresponding text are misleading; the tests weren't skipped.
```suggestion
if (failedAsExpected > 0) print(color.cyan(`  ${failedAsExpected} expected failures`));
```